monai/bundle/reference_resolver.py

```diff
@@ -172,6 +174,39 @@ def get_resolved_content(self, id: str, **kwargs):
         """
         return self._resolve_one_item(id=id, **kwargs)
 
+    @classmethod
+    def resolve_relative_ids(cls, id: str, value: str) -> str:
+        """
+        Resolve the relative reference ids to absolute ids. For example:
+
+        .. code-block:: python
+
+            {
+                "A": 1,
+                "B": "@#A",
+                "C": {"key": "@##A", "value1": 2, "value2": "@#value1", "value3": [3, 4, "@#1"]},
+            }
+
+        Args:
+            id: id name for current config item to compute relative id.
+            value: input value to resolve relative ids.
+
+        """
+        # get the prefixes like: "@####", "@###", "@#"
+        prefixes = sorted(set().union(cls.relative_id_prefix.findall(value)), reverse=True)
+        current_id = id.split(cls.sep)
+
+        for p in prefixes:
+            length = len(p[len(cls.ref) :]) // len(cls.sep)
```

Comment:
this could be substring counting: `p[len(cls.ref) :].count(cls.sep)`

Reply:
Updated in the latest commit.

Thanks.